src/base/static/components/molecules/tag-editor.js

```diff
@@ -58,28 +66,74 @@ class TagEditor extends Component {
     this._isMounted = false;
   }
 
+  onClickCreatePlaceTag = async () => {
+    const response = await mapseedApiClient.placeTags.create(
+      this.props.placeUrl,
+      {
+        note: this.state.note,
+        place: this.props.placeUrl,
+        tag: this.props.tag.url,
+      },
+    );
+
+    if (response) {
+      this.props.createPlaceTag(this.props.placeId, response);
+    } else {
+      alert("Oh dear. It looks like that didn't save. Please try again.");
+      Util.log("USER", "place-tags", "fail-to-create-place-tag");
+    }
+  };
+
+  onClickRemovePlaceTag = async () => {
+    const response = await mapseedApiClient.placeTags.delete(
+      this.props.placeTag.url,
+    );
+
+    if (response) {
+      this.props.removePlaceTag(this.props.placeId, this.props.placeTag.id);
+    } else {
+      alert("Oh dear. It looks like that didn't save. Please try again.");
+      Util.log("USER", "place-tags", "fail-to-remove-place-tag");
```

Comment:
See my earlier comment, where if the client returns a non-200, it should return `null` instead of throwing an error. If the response is not 200, I believe the app will crash with an error.